tests/compose/test_compose_sh.py

```diff
@@ -14,16 +14,31 @@
 ## See the License for the specific language governing permissions and
 ## limitations under the License.
 ## ---------------------------------------------------------------------------
-"""Test the compose-sh functionality
+"""Integration test for the compose-sh script
 """
-from alyeska.compose.compose_sh import init_flags, convert_yaml_to_sh
+import os
+import pathlib
 
-from test_compose_globals import COMPOSE_SMALL, COMPOSE_BIG, COMPOSE_CYCLE
+import pytest
 
+import alyeska.compose.compose_sh as compose_sh
 
-def test__convert_yaml_to_sh():
-    # The config_file value doesn't matter since we're passing it in explicitly,
-    # but we don't want to check for task file presence since they definitely don't exist.
-    init_flags(["foo", "--no-check"])
-    actual = convert_yaml_to_sh(COMPOSE_SMALL)
-    assert isinstance(actual, str)
+from test_compose_globals import COMPOSE_SMALL
+
+
+# Makes sure the output file is in this script's directory.
+OUTFILE: str = str(pathlib.Path(os.path.dirname(__file__), "out.sh"))
+
+
+@pytest.fixture()
+def cleanup_output():
+    yield
+    if os.path.exists(OUTFILE):
+        os.remove(OUTFILE)
+
+
+@pytest.mark.usefixtures("cleanup_output")
+def test__main():
+    # Setting --no-check since we don't want to check for task file presence. They definitely don't exist.
+    actual = compose_sh.main([str(COMPOSE_SMALL), "-o", OUTFILE, "--no-check"])
+    assert os.path.exists(OUTFILE)
```

Comment:
Do we need a fixture here? Why not just remove the file (if it exists) as part of test__main?

```suggestion
def test__main():
    # Makes sure the output file is in this script's directory.
    outfile_path = pathlib.Path(__file__).parent / "out.sh"
    if outfile_path.exists():
        outfile_path.unlink()

    # .resolve() for absolute path; easier to debug if there's an issue
    outfile_str = outfile_path.resolve().as_posix()

    # Setting --no-check since we don't want to check for task file presence. 
    # They definitely don't exist.
    compose_sh.main([str(COMPOSE_SMALL), "-o", outfile_str, "--no-check"])

    assert outfile_path.exists()
```

I rewrote the file logic with `pathlib`. `pathlib` is used pretty ubiquitously across alyeska to circumnavigate some of the unexpected, OS-specific behavior of the `os` module.

Reply:
That suggestion did not work at all how I expected. :(

Reply:
Got rid of the os usage, but kept the fixture.

Test harnesses like pytest and junit offer post-test features for a reason. They reduce boilerplate if we want to write more tests and they guarantee execution regardless of the result. I'd agree that we can get away with not using it in this case, but it's a good practice to enforce across the board.